### PR TITLE
Veoss 391

### PR DESCRIPTION
**Pull request by warrenskiwork1979**

### CODE REVIEW

**Comment by DevAndyroo** [comment]
retest this please

File: app/controllers/actions/AuthenticatedControllerComponents.scala [hunk 1]
```diff
@@ -42,10 +43,13 @@ trait AuthenticatedControllerComponents extends MessagesControllerComponents {
     auth andThen getRegistration
 
   def authAndGetOptionalData(period: Period): ActionBuilder[OptionalDataRequest, AnyContent] =
-    auth andThen getRegistration andThen getData(period)
+    auth andThen getRegistration andThen getData(period) andThen checkReturn
 
   def authAndGetData(period: Period): ActionBuilder[DataRequest, AnyContent] =
     authAndGetOptionalData(period) andThen requireData
+
+  def authAndGetOptionalDataNoReturnCheck(period: Period): ActionBuilder[OptionalDataRequest, AnyContent] =
```
Comment (darbyo) [inline app/controllers/actions/AuthenticatedControllerComponents.scala:51]: I don't think we actually need this one. I might be wrong but I don't believe any of the previous return pages actually require anything from userAnswers. Therefore don't need 'getData' we could just use authAndGetRegistration and delete this method? I'm just looking the previous pages now...

File: app/controllers/actions/CheckReturnsFilterImpl.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import scala.concurrent.{ExecutionContext, Future}
+
+class CheckReturnsFilterImpl @Inject()(connector: VatReturnConnector)(implicit val executionContext: ExecutionContext)
+
+  extends CheckReturnsFilter {
+  
+  override protected def filter[A](request: OptionalDataRequest[A]): Future[Option[Result]] = {
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
+
+    connector.get(Period(2021, Q3)) map {
```
Comment (darbyo) [inline app/controllers/actions/CheckReturnsFilterImpl.scala:40]: Can we make period dynamic here, it is in scope where this method is used so should be a quick change and will save us from falling over down the line.

**Review by darbyo — APPROVED** [review]

**Review by darbyo — CHANGES_REQUESTED** [review]
Approved by mistake, but please see comments above

File: app/controllers/actions/AuthenticatedControllerComponents.scala (hunk 1 above)
Comment (warrenskiwork1979) [reply app/controllers/actions/AuthenticatedControllerComponents.scala:51]: Have removed redundant action.

File: app/controllers/actions/CheckReturnsFilterImpl.scala (hunk 2 above)
Comment (warrenskiwork1979) [reply app/controllers/actions/CheckReturnsFilterImpl.scala]: Have made period dynamic.

**Review by darbyo — APPROVED** [review]

**Pull request merged**